pkg/southbound/gnmi/client.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+	baseClient "github.com/openconfig/gnmi/client"
+	gclient "github.com/openconfig/gnmi/client/gnmi"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// Client gNMI client interface
+type Client interface {
+	Capabilities(ctx context.Context, r *gpb.CapabilityRequest) (*gpb.CapabilityResponse, error)
+	CapabilitiesWithString(ctx context.Context, request string) (*gpb.CapabilityResponse, error)
+	Get(ctx context.Context, r *gpb.GetRequest) (*gpb.GetResponse, error)
+	GetWithString(ctx context.Context, request string) (*gpb.GetResponse, error)
+	Set(ctx context.Context, r *gpb.SetRequest) (*gpb.SetResponse, error)
+	SetWithString(ctx context.Context, request string) (*gpb.SetResponse, error)
+	Subscribe(ctx context.Context, q baseClient.Query) error
+	Close() error
```

Comment:
The interface should extend `io.Closer` instead of defining a `Close() error` method

Reply:
Done